### PR TITLE
Switching zrangethenrem and zrangebyscorethenrem to lua scripts

### PR DESCRIPTION
**Pull request by v-kat**
For moving off forked redis and going to elasticache.
Once this is deployed we can move off the forked redis and redis_cluster clients. After that it's data migration then moving off the forked redis.

Some help testing this locally except unit tests would be nice @chongbomb 

Same number of failing tests 496 examples, 29 failures

### CODE REVIEW

**Review by tarpdalton — APPROVED** [review]

File: lib/rpush/daemon/store/redis.rb [hunk 1]
```diff
@@ -120,6 +120,56 @@ def translate_integer_notification_id(id)
 
         private
 
+        ZRANGEBYSCORETHENREM_SCRIPT = <<-EOS
+          -- Redis lua script
+          local queue_key = KEYS[1]
+          local min = ARGV[1]
+          local max = ARGV[2]
+          redis.call('zrangebyscore', queue_key, min, max)
+          local return_results = redis.call('zremrangebyscore', queue_key, min, max)
```
Comment (cibingeorge) [inline lib/rpush/daemon/store/redis.rb]: this doesnt make any sense to me. `zrangebyscore` is a getter. But we are not using the returned values anywhere. Whats the point of calling `zrangebyscore` ?

File: lib/rpush/daemon/store/redis.rb [hunk 2]
```diff
@@ -120,6 +120,56 @@ def translate_integer_notification_id(id)
...
+          result
+        end
+
+        ZRANGETHENREM_SCRIPT = <<-EOS
+          -- Redis lua script
+          local queue_key = KEYS[1]
+          local start = ARGV[1]
+          local stop = ARGV[2]
+          redis.call('zrange', queue_key, start, stop)
+          local return_results = redis.call('zremrangebyrank', queue_key, start, stop)
```
Comment (cibingeorge) [inline lib/rpush/daemon/store/redis.rb]: same as above. `zrange` is a getter. Theres no point in calling it

File: lib/rpush/daemon/store/redis.rb (hunk 1 above)
Comment (chongbomb) [reply lib/rpush/daemon/store/redis.rb]: The score is the epoch time the notification is created.

File: lib/rpush/daemon/store/redis.rb [hunk 3]
```diff
@@ -120,6 +120,56 @@ def translate_integer_notification_id(id)
...
+          end
+          result
+        end
+
+        ZRANGETHENREM_SCRIPT = <<-EOS
+          -- Redis lua script
+          local queue_key = KEYS[1]
+          local start = ARGV[1]
+          local stop = ARGV[2]
+          redis.call('zrange', queue_key, start, stop)
```
Comment (chongbomb) [inline lib/rpush/daemon/store/redis.rb]: This should be the return value.

File: lib/rpush/daemon/store/redis.rb [hunk 4]
```diff
@@ -120,6 +120,56 @@ def translate_integer_notification_id(id)
 
         private
 
+        ZRANGEBYSCORETHENREM_SCRIPT = <<-EOS
+          -- Redis lua script
+          local queue_key = KEYS[1]
+          local min = ARGV[1]
+          local max = ARGV[2]
+          redis.call('zrangebyscore', queue_key, min, max)
```
Comment (chongbomb) [inline lib/rpush/daemon/store/redis.rb]: This should be the return value.

**Review by chongbomb — CHANGES_REQUESTED** [review]

**Comment by cibingeorge** [comment]
This looks good, lets merge this in?

**Pull request merged**